### ISSUE TITLE
Is there a provision to add custom input parameters in User Message constructor?

### ISSUE THREAD

**Issue by shatacheruvu** [RasaHQ/rasa_core#674]
**Rasa Core version**: 0.9.6

**Python version**: 3.5.6

**Operating system** (windows, osx, ...): OSX

**Aim**
We want to share the bot's capabilities across one multi-tenant system. The bot remains common to every user who is segregated based on roles. The main problem here is about authentication and authorisation of the user talking to the bot. 

**Issue**:
The current input channel primarily accepts only `sender_id` and `message`. There is no way of initialising a conversation with default slot values(For instance: `user_role`). How do we pass custom fields in the `User Message(...)` constructor apart from the ones mentioned above?

**Comment by amn41** [comment]
It's an interesting suggestion.

The way to do this currently is to define a slot for this (e.g. user_role).

For example, if you have an action `ActionFetchProfile`, it can `return [SlotSet("role": "admin")]`  or similar. 

What's a use case where you would need to alter the message object itself?

**Comment by shatacheruvu** [comment]
The action `ActionFetchProfile` will return `admin` only if it gets user id. The issue is that there is no provision to send user id explicitly along with `sender_id` and `message`. 

**Scenario:**
The engine interprets messages from a chat front-end. So, the `sender_id` is primarily the group/channel id. The context in Rasa is maintained Group wise. The `user_id` is mainly required to restrict responses in a group or individually based on his/her authorisation level.

**Comment by amn41** [comment]
would it not be possible to include the user id in the sender id? e.g. `sender_id = user_id+session_id`?

**Comment by shatacheruvu** [comment]
Yeah, technically it is possible. Although, that is one lengthy work around because we will have to maintain a slot for user role throughout the conversation and we also may have to train stories with checkpoints based on user role. 

Apart from this issue do you think we should even consider having a single bot approach or is there a better way?

**Comment by amn41** [comment]
I'm not sure I understand your question  - could you please provide an example?

same with 'single bot approach' - I'm not sure I understand what you mean
